### PR TITLE
Add `minify` option

### LINKED ISSUE #8: [Feature request] Support minified output
Recently I am implementing "minify srcset" feature for [htmlnano](https://github.com/posthtml/htmlnano), a modular HTML minifier. And I decide to utilize `srcset` in the feature.

According to [HTML Standard](https://html.spec.whatwg.org/#srcset-attributes), both of following HTML markups are valid:

```html
<img srcset="image.png 1x, image2.png 2x, image2.png 3x">
```

```html
<img srcset="image.png 1x,image2.png 2x,image2.png 3x">
```

Currently `srcset`'s `stringify` method only supports the first one. Although I could bring up my own `stringify` function, it is better to have a built-in option for `srcset#stringify`.

### PR DESCRIPTION
**Pull request by SukkaW**
This closes #8.

The test cases, type definition, and README has been updated as well.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -51,17 +51,28 @@ console.log(stringified);
...
+- Returns: `<srcset.SrcSetDefinition>` An array of objects with the possible properties: `url` (always), `width`, `density`.
 
-### .stringify()
+### .stringify(srcSetDefinition[, stringifyOption])
 
-Accepts an array of objects with the possible properties: `url` (required), `width`, `density` and returns a srcset string.
+- `srcSetDefinition`: `<srcset.SrcSetDefinition>` Accepts an array of objects with the possible properties: `url` (required), `width`, `density`.
+- `stringifyOption`: `<srcset.StringifyOption>`
+  - `minify`: `<boolean>`: Whether to strip redundant white spaces. Default `false`.  
+    **Warning!** Enable `minify` could results in an invalid srcset string!
```
Comment (SukkaW) [inline readme.md:74]: Although most browsers will tolerate the "minified" srcset string, technically it does not fully comply with [HTML Living Standard](https://html.spec.whatwg.org/#srcset-attributes).

File: index.d.ts [hunk 2]
```diff
@@ -4,6 +4,10 @@ declare namespace srcset {
 		width?: number;
 		density?: number;
 	}
+
+	interface StringifyOption {
+		minify?: boolean;
```
Comment (sindresorhus) [inline index.d.ts:9]: This needs a doc comment

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 2 above)
Comment (SukkaW) [reply index.d.ts:9]: Should I add a `@description` here?

Comment (SukkaW) [reply index.d.ts:12]: Done. The `description` will now show on supported IDE:

image

> Screenshot from me viewing `test.js`.

File: index.js [hunk 3]
```diff
@@ -53,8 +53,8 @@ exports.parse = string => {
 	);
 };
 
-exports.stringify = array => {
-	return [...new Set(
+exports.stringify = (array, {minify} = {minify: false}) => {
```
Comment (sindresorhus) [inline index.js:56]:
```suggestion
exports.stringify = (array, {minify = false} = {}) => {
```

File: index.d.ts (hunk 2 above)
Comment (sindresorhus) [reply index.d.ts:12]: Here's how to format it: https://github.com/sindresorhus/typescript-definition-style-guide

File: index.d.ts [hunk 4]
```diff
@@ -4,6 +4,13 @@ declare namespace srcset {
 		width?: number;
 		density?: number;
 	}
+
+	interface StringifyOption {
```
Comment (sindresorhus) [inline index.d.ts:8]:
```suggestion
	interface StringifyOptions {
```

File: index.test-d.ts [hunk 5]
```diff
@@ -10,3 +10,6 @@ parsed.push({url: 'banner-phone-HD.jpg', density: 2});
 expectError(parsed.push({}));
 
 expectType<string>(srcset.stringify(parsed));
+
+expectError(srcset.stringify(parsed, true));
+expectError(srcset.stringify(parsed, {minify: 1}));
```
Comment (sindresorhus) [inline index.test-d.ts:15]: This should test the valid case.

File: readme.md [hunk 6]
```diff
@@ -51,17 +51,28 @@ console.log(stringified);
...
 
-### .parse()
+### .parse(srcset)
 
-Accepts a srcset string and returns an array of objects with the possible properties: `url` (always), `width`, `density`.
+- `srcset`: `<string>` Accepts a srcset string
+- Returns: `<srcset.SrcSetDefinition>` An array of objects with the possible properties: `url` (always), `width`, `density`.
 
-### .stringify()
+### .stringify(srcSetDefinition[, stringifyOption])
```
Comment (sindresorhus) [inline readme.md:69]:
```suggestion
### .stringify(srcSetDefinition, options?)
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Closing for lack of response.

**Pull request closed without merging**